dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/ToolTest.java

```diff
@@ -53,6 +54,37 @@ public static void teardown() throws Exception {
         server.stopServer();
     }
 
+    @Test
+    public void postJsonRpc() throws Exception {
+        String jsonBody = """
+                            {
+                                "jsonrpc": "2.0",
+                                "method": "tools.call",
+                                "id": 1
+                            }
+                        """;
+        HttpRequest request = new HttpRequest(server, "/toolTest/mcp")
+                                                                      .requestProp("Accept", "application/json")
+                                                                      .jsonBody(jsonBody)
+                                                                      .method("POST");
+
+        int statusCode = -1;
+        String contentType = null;
+        String response = null;
+        try {
+            response = request.run(String.class);
+        } catch (Exception e) {
+            statusCode = request.getResponseCode();
+            contentType = request.getResponseHeaders().get("Content-Type");
+
+            System.out.println("Expected failure due to incorrect Accept header.");
+            System.out.println("Status Code: " + statusCode);
+            System.out.println("Content-Type Header: " + contentType);
+            System.out.println("Raw Response: " + response);
+        }
```

Comment:
This will catch and hide any unexpected exceptions.

Use `HttpRequest.expectCode` to indicate that 406 is expected and then you shouldn't need to catch the exception here.